src/toxcore/onion/onion_client.rs

```diff
@@ -292,6 +294,24 @@ impl OnionClient {
         }
     }
 
+    fn is_redundant_ping(&self, pk: PublicKey, search_pk: PublicKey, request_queue: &RequestQueue<AnnounceRequestData>) -> bool {
+        let check_pks = |data: &AnnounceRequestData| -> bool {
+            let request_search_pk = if let Some(friend_pk) = data.friend_pk {
+                friend_pk
+            } else {
+                self.dht.pk
+            };
+            data.pk == pk && search_pk == request_search_pk
+        };
+        let request = request_queue.get_values()
+            .find(|(_ping_time, request_data)| check_pks(request_data));
```

Comment:
After yesterday's discussion with @kpp I realized that this is not a correct check. We have different timeouts for allowing new requests and for expiration of old requests. So it's possible that we have different entries in the queue for the same request. Which means instead of finding the first one we should check all matches, i.e. instead of `find` we should use `any` or `all`. Otherwise we can send a new request even though we sent it recently.

Reply:
done